cmf-cli/Commands/publish/PublishCommand.cs

```diff
@@ -0,0 +1,222 @@
+        public override void Configure(Command cmd)
+        {
+            cmd.AddArgument(new Argument<DirectoryInfo>(
+                name: "workingDir",
+                getDefaultValue: () => { return new("."); },
+                description: "Working Directory"));
+
+            cmd.AddOption(new Option<DirectoryInfo>(
+                aliases: new string[] { "-o", "--outputDir" },
+                getDefaultValue: () => { return new("Package"); },
+                description: "Output directory for created package"));
+
+            cmd.AddOption(new Option<string>(
+                aliases: new string[] { "-r", "--repo" },
+                description: "Repository where dependencies are located (url or folder)"));
+
+            cmd.AddOption(new Option<bool>(
+                aliases: new string[] { "-t", "--publishTests" },
+                description: "to include and publish test packages"));
+
+            // Add the handler
+            cmd.Handler = CommandHandler.Create<DirectoryInfo, DirectoryInfo, string, bool>(Execute);
+        }
+
+        /// <summary>
+        /// Executes the specified working dir.
+        /// </summary>
+        /// <param name="workingDir">The working dir.</param>
+        /// <param name="outputDir">The output dir.</param>
+        /// <param name="repo">The repo.</param>
+        /// <param name="publishTests">True to publish test packages</param>
+        /// <returns></returns>
+        public void Execute(DirectoryInfo workingDir, DirectoryInfo outputDir, string repo, bool publishTests)
+        {
+            FileInfo cmfpackageFile = new($"{workingDir}/{CliConstants.CmfPackageFileName}");
+
+            Uri repoUri = repo != null ? new Uri(repo) : null;
+
+            // Reading cmfPackage
+            CmfPackage cmfPackage = CmfPackage.Load(cmfpackageFile, setDefaultValues: true);
+
+            Execute(cmfPackage, outputDir, repoUri, publishTests);
+        }
+
+        /// <summary>
+        /// Executes the specified CMF package.
+        /// </summary>
+        /// <param name="cmfPackage">The CMF package.</param>
+        /// <param name="outputDir">The output dir.</param>
+        /// <param name="repoUri">The repo URI.</param>
+        /// <param name="publishTests">True to publish test packages</param>
+        /// <returns></returns>
+        /// <exception cref="CmfPackageCollection">
+        /// </exception>
+        public void Execute(CmfPackage cmfPackage, DirectoryInfo outputDir, Uri repoUri, bool publishTests)
+        {
+            #region Output Directories Handling
+
+            DirectoryInfo packageDirectory = cmfPackage.GetFileInfo().Directory;
+            outputDir = FileSystemUtilities.GetOutputDir(cmfPackage, outputDir, true);
+            if (outputDir == null)
+            {
+                return;
+            }
+
+            DirectoryInfo packageOutputDir = FileSystemUtilities.GetPackageOutputDir(cmfPackage, packageDirectory);
```

Comment:
This doesn't appear to be used, except in the `finally`. Do we really need to do this?

Reply:
Fixed